R/utils.R

```diff
@@ -1862,3 +1862,147 @@ find_rank_bias <- function(mat,
+    
+    rownames(zeroExpressionMatrix) <- missing_rows 
+    #Assign row names
+    colnames(zeroExpressionMatrix) <- colnames(ref_matrix) 
+    #Assign column names
+    
+    full_matrix <- rbind(ref_matrix, zeroExpressionMatrix) 
+    #Bind GSEMatrix and zeroExpressionMatrix together
+    
+    #Reorder matrix
+    full_matrix <- full_matrix[gene_vector, ] 
+    #Reorder fullMatrix to preserve gene order
+    return(full_matrix) 
+    #Return fullMatrix
+}
+
+#' Given a count matrix, determine if the matrix has been either 
+#' log-normalized, normalized, or contains raw counts
+#' @param counts_matrix Count matrix containing scRNA-seq read data
+#' @param max_log_value Static value to determine if a matrix is normalized
+#' @return String either raw counts, log-normalized or normalized
+#' @example
+#' mat <- append_genes(
+#' gene_vector = human_genes_10x,
+#' ref_matrix = cbmc_ref 
+#' ) 
+#' @export
+check_raw_counts <- function(counts_matrix, max_log_value = 50)
+{
+    if(!is.matrix(counts_matrix))
+    {
+        GSEMatrix <- as.matrix(counts_matrix)
+    }
+    if (is.integer(counts_matrix))
+    {
+        return("raw counts")
+    }
+    else if (is.double(counts_matrix))
+    {
+        if (all(counts_matrix == floor(counts_matrix)))
+        {
+            return("raw counts")
+        }
+        if(max(counts_matrix) > max_log_value)
+        {
+            return("normalized")
+        }
+        else if (min(counts_matrix) < 0)
+        {
+            stop("negative values detected, likely scaled data")
+        }
+        else
+        {
+            return("log-normalized")
+        }
+    }
+    else
+    {
+        stop("unknown matrix format: ", typeof(counts_matrix))
+    }
+}
+
+#' Function to combine records into single atlas
+#' @param matrix_fns character vector of paths to study matrices stored as .rds files.
+#' If a named character vector, then the name will be added as a suffix to the cell type
+#' name in the final matrix. If it is not named, then the filename will be used (without .rds)
+#' @param genes_fn text file with a single column containing genes and the ordering desired
+#' in the output matrix
+#' @param output_fn output filename for .rds file. If NULL the matrix will be returned instead of
+#' saving
+#' @example
+#' 
+#' @export
+build_atlas <- function(matrix_fns,
+                        genes_fn,
+                        output_fn = NULL){
+    
+    genesVector <- readr::read_lines(genes_fn)
+    
+    ref_mats <- lapply(matrix_fns, readRDS)
+    
+    if(is.null(names(matrix_fns))){
+        names(ref_mats) <- basename(matrix_fns) %>% stringr::str_remove(".rds$")
+    } else {
+        names(ref_mats) <- names(matrix_fns)
+    }
+    
+    # iterate over list and get new matrices
+    new_mats <- list()
+    for(i in seq_along(ref_mats)){
+        # standardize genes in matrix
+        mat <- clustifyr:::append_genes(gene_vector = genesVector,
```

Comment:
no longer need "clustifyr:::", sorry for the confusion

Reply:
See f20602a